### PR TITLE
Hw 19 1

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

File: src/main/java/core/basesyntax/db/Storage.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.model.Manufacturer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    public static final List<Driver> drivers = new ArrayList<>();
```
Comment (AndreiYazadzhi) [inline src/main/java/core/basesyntax/db/Storage.java]: Do you need drivers list in our case?

File: src/main/java/core/basesyntax/db/Storage.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public static final List<Manufacturer> manufacturers = new ArrayList<>();
+    public static final List<Car> cars = new ArrayList<>();
```
Comment (AndreiYazadzhi) [inline src/main/java/core/basesyntax/db/Storage.java]: Same with cars

File: src/main/java/core/basesyntax/db/Storage.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private static long driverId = 0;
```
Comment (AndreiYazadzhi) [inline src/main/java/core/basesyntax/db/Storage.java:13]: Redundant variable

File: src/main/java/core/basesyntax/db/Storage.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    private static long manufacturerId = 0;
+    private static long carId = 0;
```
Comment (AndreiYazadzhi) [inline src/main/java/core/basesyntax/db/Storage.java]: Same

File: src/main/java/core/basesyntax/db/Storage.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        manufacturers.add(manufacturer);
+        return manufacturer;
+    }
+
+    public static Car addCar(Car car) {
+        carId++;
+        car.setId(carId);
+        cars.add(car);
+        return car;
+    }
```
Comment (AndreiYazadzhi) [inline src/main/java/core/basesyntax/db/Storage.java]: Redundant method

File: src/main/java/core/basesyntax/db/Storage.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    private static long driverId = 0;
+    private static long manufacturerId = 0;
+    private static long carId = 0;
+
+    public static Driver addDriver(Driver driver) {
+        driverId++;
+        driver.setId(driverId);
+        drivers.add(driver);
+        return driver;
+    }
```
Comment (AndreiYazadzhi) [inline src/main/java/core/basesyntax/db/Storage.java]: Redundant method

**Review by AndreiYazadzhi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/db/Storage.java (hunk 5 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/db/Storage.java]: done

File: src/main/java/core/basesyntax/db/Storage.java (hunk 6 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/db/Storage.java]: done

File: src/main/java/core/basesyntax/db/Storage.java (hunk 4 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/db/Storage.java]: done

File: src/main/java/core/basesyntax/db/Storage.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/db/Storage.java]: done

File: src/main/java/core/basesyntax/db/Storage.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/db/Storage.java]: done, but I think today I will add these lines back )

**Review by AndreiYazadzhi — APPROVED** [review]

File: pom.xml [hunk 7]
```diff
@@ -8,5 +8,61 @@
...
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
+        <maven.checkstyle.plugin.configLocation>
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
```
Comment (Sofasmile) [inline pom.xml]: do we need this dependency?

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer manufacturerToUpdate = Storage.manufacturers.stream()
+                .filter(m -> m.getId().equals(manufacturer.getId()))
+                .findFirst().get();
+        manufacturerToUpdate.setCountry(manufacturer.getCountry());
+        manufacturerToUpdate.setName(manufacturer.getName());
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        return manufacturerToUpdate;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        for (int i = 0; i < Storage.manufacturers.size(); i++) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: read about method `removeIf`

File: src/main/java/core/basesyntax/db/Storage.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.Manufacturer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    public static final List<Manufacturer> manufacturers = new ArrayList<>();
+    private static long manufacturerId = 0;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/db/Storage.java]:
```suggestion
    private static Long manufacturerId = 0;
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 7 above)
Comment (semyonich) [reply pom.xml]: removed

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java (hunk 8 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: done

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java (hunk 9 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: done

File: src/main/java/core/basesyntax/db/Storage.java (hunk 10 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/db/Storage.java]: done

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java [hunk 11]
```diff
@@ -27,22 +27,17 @@ public Manufacturer create(Manufacturer manufacturer) {
...
-        Manufacturer manufacturerToUpdate = Storage.manufacturers.stream()
-                .filter(m -> m.getId().equals(manufacturer.getId()))
-                .findFirst().get();
-        manufacturerToUpdate.setCountry(manufacturer.getCountry());
-        manufacturerToUpdate.setName(manufacturer.getName());
-        return manufacturerToUpdate;
+        for (int i = 0; i < Storage.manufacturers.size(); i++) {
+            if (Storage.manufacturers.get(i).getId().equals(manufacturer.getId())) {
+                Storage.manufacturers.set(i, manufacturer);
+                return Storage.manufacturers.get(i);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]:
```suggestion
                break;
```

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java [hunk 12]
```diff
@@ -27,22 +27,17 @@ public Manufacturer create(Manufacturer manufacturer) {
...
+            }
+        }
+        return Storage.addManufacturer(manufacturer);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]:
```suggestion
        return manufacturer;
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]
good job)

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java (hunk 11 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: done

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java (hunk 12 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**